core/commands/dht.go

```diff
@@ -307,7 +307,7 @@ var provideRefDhtCmd = &cmds.Command{
 		ctx := notif.RegisterForQueryEvents(req.Context(), events)
 
 		go func() {
-			defer close(outChan)
+			closeUnlessErr(ctx, outChan)
```

Comment:
defer

Reply:
Done, thanks!

I *think* I was also using the wrong context, so I've changed `ctx` to `req.Context()`.